apps/tx-builder/src/components/TransactionDetails.tsx

```diff
@@ -0,0 +1,176 @@
+                  {/* param name */}
+                  <StyledMethodNameLabel size="xl" color="secondaryHover" tooltip={methodLabel}>
+                    {methodLabel}
+                  </StyledMethodNameLabel>
+                  {/* param value */}
+                  <TxValueLabel>{contractFieldsValues?.[methodName]}</TxValueLabel>
+                </React.Fragment>
+              );
+            })}
+          </>
+        )}
+      </TxSummaryContainer>
+    </Wrapper>
+  );
+};
+
+export default TransactionDetails;
+
+const Wrapper = styled.article`
+  flex-grow: 1;
+  padding: 0 16px;
+  user-select: text;
+`;
+
+const TxSummaryContainer = styled.div`
+  display: grid;
+  grid-template-columns: minmax(100px, 2fr) minmax(100px, 5fr);
+  gap: 4px;
+
+  margin-top: 16px;
+`;
+
+const StyledTxTitle = styled(Title)`
+  font-size: 16px;
+  margin: 8px 0;
+  font-weight: bold;
+  line-height: initial;
+`;
+
+const StyledMethodNameLabel = styled(Text)`
+  padding-left: 4px;
+
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+const LINE_HEIGHT = 22;
+const MAX_HEIGHT = 2 * LINE_HEIGHT; // 2 lines as max height
+
+const TxValueLabel = ({ children }: { children: React.ReactNode }) => {
+  const [showMore, setShowMore] = useState(false);
+  const [showEllipsis, setShowEllipsis] = useState(false);
+  const valueContainerRef = useRef<HTMLDivElement>(null);
+
+  const containerHeight = useElementHeight(valueContainerRef);
+
+  // we show the Show more/less button if the height is more than 44px (the height of 2 lines)
+  const showMoreButton = containerHeight && containerHeight > MAX_HEIGHT;
+
+  // we show/hide ellipsis at the end of the second line if user clicks on "Show more"
+  useEffect(() => {
+    if (showMoreButton && !showMore) {
+      setShowEllipsis(true);
+    }
+  }, [showMoreButton, showMore]);
+
+  return (
+    <div ref={valueContainerRef}>
+      {/* value */}
+      <StyledTxValueLabel size="xl" showMore={showMore} showEllipsis={showEllipsis}>
+        {children}
+      </StyledTxValueLabel>
+
+      {/* show more/less button */}
+      {showMoreButton && (
+        <StyledButtonLink color="primary" onClick={() => setShowMore((showMore) => !showMore)}>
+          {showMore ? 'Show less' : 'Show more'}
+        </StyledButtonLink>
+      )}
+    </div>
+  );
+};
+
+const StyledTxValueLabel = styled(Text).withConfig({
+  shouldForwardProp: (prop) => !['showMore'].includes(prop) || !['showEllipsis'].includes(prop),
+})<{ showMore?: boolean; showEllipsis?: boolean }>`
+  max-height: ${({ showMore }) => (showMore ? '100%' : `${MAX_HEIGHT + 1}px`)};
+
+  line-break: anywhere;
+  overflow: hidden;
+  word-break: break-all;
+  text-overflow: ellipsis;
+
+  ${({ showEllipsis, showMore }) =>
+    !showMore &&
+    showEllipsis &&
+    `@supports (-webkit-line-clamp: 2) {
+      display: -webkit-box;
+      -webkit-line-clamp: 2;
+      -webkit-box-orient: vertical;
+  }`}
```

Comment:
see https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp
for more info about `-webkit-line-clamp`